src/main/java/com/rackspace/ceres/app/config/configValidator/ConcreteGranularityValidator.java

```diff
@@ -0,0 +1,43 @@
+package com.rackspace.ceres.app.config.configValidator;
+
+import com.rackspace.ceres.app.config.DownsampleProperties;
+import com.rackspace.ceres.app.config.DownsampleProperties.Granularity;
+import java.util.List;
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+import org.springframework.beans.BeanWrapperImpl;
+
+public class ConcreteGranularityValidator implements ConstraintValidator<GranularityValidator, List<Granularity>> {
+
+  public void initialize() { }
+
+  @Override
+  public boolean isValid(List<Granularity> granularityList, ConstraintValidatorContext constraintValidatorContext) {
+    if(granularityList == null) {
+      return true;
+    }
+
+    granularityList.sort((g1, g2) -> g1.getWidth().compareTo(g2.getWidth()));
+
+    for(int i = 0; i < granularityList.size()-1; i++) {
+      if(i != granularityList.size() && granularityList.get(i+1).getWidth().getSeconds() % granularityList.get(i).getWidth().getSeconds() != 0) {
```

Comment:
Line is too long. 120 char is advisable. Also do you really need i != granularityList.size()? you already have size() -1 in the for header.